File: generate-qr.js
// QR Code Generator for Hostel Feedback Project
const https = require('https');
const fs = require('fs');
const path = require('path');

// URL that the QR code will link to
const url = 'http://localhost:8080/index.html';
const size = 400; // QR code size in pixels

// Encode the URL for the API
const encodedUrl = encodeURIComponent(url);

// QR code API endpoint
const apiUrl = `https://api.qrserver.com/v1/create-qr-code/?size=${size}x${size}&data=${encodedUrl}&margin=10&format=png`;

// Output path
const outputPath = path.join(__dirname, 'frontend', 'hostel-feedback-qr.png');
const outputDir = path.dirname(outputPath);

// Create frontend directory if it doesn't exist
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

console.log('🔄 Generating QR Code...');
console.log(`🔗 URL: ${url}`);
console.log(`📏 Size: ${size}x${size} pixels`);

// Download QR code image
https.get(apiUrl, (response) => {
  if (response.statusCode === 200) {
    const fileStream = fs.createWriteStream(outputPath);
    response.pipe(fileStream);
    
    fileStream.on('finish', () => {
      fileStream.close();
      console.log('✅ QR Code generated successfully!');
      console.log(`📁 Saved to: ${outputPath}`);
      console.log(`📱 Scan this QR code to open: ${url}`);
      console.log('✨ QR code image is ready to use!');
    });

    fileStream.on('error', (err) => {
      fs.unlink(outputPath, () => {}); // Delete file on error
      console.error(`❌ Error saving file: ${err}`);
      process.exit(1);
    });
  } else {
    console.error(`❌ Failed to download QR code: HTTP ${response.statusCode}`);
    process.exit(1);
  }
}).on('error', (err) => {
  console.error(`❌ Error: ${err}`);
  process.exit(1);
});

File: generate_qr.py
#!/usr/bin/env python3
"""
QR Code Generator for Hostel Feedback Project
Generates a QR code image that links to the index page
"""

import qrcode
import os
from pathlib import Path

def generate_qr_code():
    # URL that the QR code will link to
    url = "http://localhost:8080/index.html"
    
    # Create QR code instance
    qr = qrcode.QRCode(
        version=1,
        error_correction=qrcode.constants.ERROR_CORRECT_H,
        box_size=10,
        border=4,
    )
    
    # Add data and optimize
    qr.add_data(url)
    qr.make(fit=True)
    
    # Create an image from the QR code
    img = qr.make_image(fill_color="black", back_color="white")
    
    # Define output path
    output_dir = Path(__file__).parent / "frontend"
    output_path = output_dir / "hostel-feedback-qr.png"
    
    # Create frontend directory if it doesn't exist
    output_dir.mkdir(parents=True, exist_ok=True)
    
    # Save the image
    img.save(output_path)
    
    print(f"✅ QR Code generated successfully!")
    print(f"📁 Saved to: {output_path}")
    print(f"🔗 URL encoded: {url}")
    print(f"📱 Scan this QR code to open: {url}")
    
    return output_path

if __name__ == "__main__":
    try:
        output_file = generate_qr_code()
        print(f"\n✨ You can now use '{output_file}' in your project!")
    except ImportError:
        print("❌ Error: 'qrcode' module not installed")
        print("📦 Install it with: pip install qrcode[pil]")
    except Exception as e:
        print(f"❌ Error generating QR code: {e}")
